### PR TITLE
Added Modals for QnA and refactored api endpoint

### PR DESCRIPTION
**Pull request by ashequl93**
This PR contains a couple of changes.
1. I have added the Higher Order Component to handle _**POST /interactions**_ endpoint
2. Refactored Questions & Answers endpoint to retrieve all the questions from multiple pages using async/await
3. Added Modals for Add New Questions and Add Answer
Screen Shot 2021-09-22 at 2 50 00 PM
Screen Shot 2021-09-22 at 2 50 10 PM

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ module.exports = {
   },
   parserOptions: {
     'sourceType': 'module',
+    'ecmaVersion': 8,
```
Comment (Jhaines1988) [inline .eslintrc.js:13]: What  is this doing for us ?  Not a high priority just curios

File: client/src/components/HOC/withInteractionApi.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+var withInteractionsApi = function (WrappedComponent, widget) {
+
+  return class extends React.Component {
+    constructor(props) {
+      super(props);
+      this.sendInteraction = this.sendInteraction.bind(this);
+      this.formatBody = this.formatBody.bind(this);
+    }
+
+    formatBody(method, apiRoute, params = {}, data = {}) {
```
Comment (Jhaines1988) [inline client/src/components/HOC/withInteractionApi.jsx:13]: As discussed we may have to change this, but Thanks so much for taking this task on. Much appreciated Ash!

File: client/src/components/QnA/Modals/AddNewAnswer.css [hunk 3]
```diff
@@ -0,0 +1,99 @@
+.answer-modal {
```
Comment (Jhaines1988) [inline client/src/components/QnA/Modals/AddNewAnswer.css:1]: Awesome to see an example of a successful modal. I may  have to do such a thing for the images. Very cool

File: client/src/components/QnA/QuestionsList.jsx [hunk 4]
```diff
@@ -20,22 +25,71 @@ class QuestionsList extends React.Component {
...
+          return {
+            questionList: [...results.data.results]
+          };
+        }, () => {
+          // call api to get all the questions of that particular product
+          let questionListResult = [];
+          const getAllQuestions = async () => {
+            let pageCount = 1;
+
+            while (true) {
```
Comment (Jhaines1988) [inline client/src/components/QnA/QuestionsList.jsx:70]: Interesting function, Ash.   Does  " while(true) "  mean while there are still pages being returned?

File: package.json [hunk 5]
```diff
@@ -66,6 +66,7 @@
     "webpack-cli": "^4.8.0"
   },
   "devDependencies": {
+    "@babel/plugin-transform-runtime": "^7.15.0",
```
Comment (Jhaines1988) [inline package.json:69]: What is this doing ?   Not urgent for approval, just curious what impact this has on our project

File: .eslintrc.js (hunk 1 above)
Comment (ashequl93) [reply .eslintrc.js:13]: I have added async/await to get all the questions across multiple pages. Using async/await needs ECMAScript version 8 to work.

**Review by Jhaines1988 — APPROVED** [review]
Looks good,  get back to me with comment questions whenever you can. Thanks for taking this on Ash!

File: client/src/components/HOC/withInteractionApi.jsx (hunk 2 above)
Comment (ashequl93) [reply client/src/components/HOC/withInteractionApi.jsx:13]: No problem. It's my pleasure.

File: client/src/components/QnA/Modals/AddNewAnswer.css (hunk 3 above)
Comment (ashequl93) [reply client/src/components/QnA/Modals/AddNewAnswer.css:1]: Thanks. Doing the CSS for Modal gave me so much pain.

File: client/src/components/QnA/QuestionsList.jsx (hunk 4 above)
Comment (ashequl93) [reply client/src/components/QnA/QuestionsList.jsx:70]: yes. As the api doesn't tell across how many pages it has the results, I need to run this logic until any page returns empty list of questions.

File: client/src/components/QnA/QuestionsList.jsx [hunk 6]
```diff
@@ -20,22 +25,71 @@ class QuestionsList extends React.Component {
...
+  }
+
   componentDidMount() {
 
     // Destructing props
     const {formatBody} = this.props;
     const params =
      {
-       product_id: 47425,
+       product_id: 47422,
```
[the remark this answers is not in the document]
Comment (ashequl93) [reply client/src/components/QnA/QuestionsList.jsx:49]: I will change it later to reflect the default state.

File: package.json (hunk 5 above)
Comment (ashequl93) [reply package.json:69]: I needed this for babel to work with async/await.

**Review by aaronfife — APPROVED** [review]

**Pull request merged**